app/services/catalog/copy_portfolio.rb

```diff
@@ -18,6 +18,10 @@ def process
     def make_copy
       @portfolio.dup.tap do |new_portfolio|
         new_portfolio.name = @name || Catalog::NameAdjust.create_copy_name(@portfolio.name, Portfolio.all.pluck(:name))
+        if @portfolio.icon_id.present?
```

Comment:
I wonder if we can move this behavior into a shared location? The code for the `portfolio` looks very close to the that of the `portfolio_item`.

Possibly something like `duplicate_icon if object.icon_id.present?`

Reply:
Moved it to a mixin that handles the duplicating of icons!